### ISSUE TITLE
Exclude @ symbol check at value start

### ISSUE THREAD

**Issue by Dum6o** [adrienverge/yamllint#531]
I have used linter to lint spring-boot framework `application.yaml` file:
``` yaml
---
spring:
  application:
    name: @project.name@
```
`@project.name@` value allows spring-boot application to resolve application name from `pom.xml` file.
However when I run linter I get an error:
`Error: ./src/main/resources/application.yaml:4:11: [error] syntax error: found character '@' that cannot start any token (syntax)`

---

**Is there any rule how can I exclude special characters check?**

**Comment by adrienverge** [comment]
Hello,

It would need to be checked in the [YAML specification](https://yaml.org/spec/1.2.2/), but I think `name: @project.name@` is **not** valid YAML.

Could this *Spring Boot format* be a customized language, needing to be pre-processed before being YAML? It would be similar to Jinja2, or GitHub workflow languages: any YAML parser would fail to interpret such a file. See #61, #65, #128, #311, #408, #460, #462, #507...

In your specific case, maybe using quotes would work (`name: "@project.name@"`).

**Comment by perlpunk** [comment]
Correct, `@` is a [reserved character](https://yaml.org/spec/1.2.2/#rule-c-reserved) and cannot start a plain scalar.
